_ml-commons-plugin/connectors.md

```diff
@@ -0,0 +1,459 @@
+
+Additional connectors will be added to this page as they are tested and verified. 
+
+
+## Prerequisites
+
+If you are an admin deploying an ML connector, make sure that the target model of the connector has already been deployed on your chosen platform. Furthermore, make sure that you have permissions to send and receive data to the third-party API for your connector. 
+
+When access control is enabled on your third-party platform, you can enter your security settings using the `authorization` or `credential` settings inside the connector API.
+
+### Adding trusted endpoints
+
+To configure connectors in OpenSearch, add the trusted endpoints to your cluster settings using the `plugins.ml_commons.trusted_connector_endpoints_regex` setting, which supports Java regex expressions, as shown in the following example:
+
+```json
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.trusted_connector_endpoints_regex": [
+            "^https://runtime\\.sagemaker\\..*\\.amazonaws\\.com/.*$",
+            "^https://api\\.openai\\.com/.*$",
+            "^https://api\\.cohere\\.ai/.*$",
+            "^https://bedrock\\..*\\.amazonaws.com/.*$"
+        ]
+    }
+}
+```
+{% include copy-curl.html %}
+
+### Enabling ML nodes
+
+Most connectors require the use of dedicated ML nodes. To make sure you have ML nodes enabled, update the following cluster settings:
+
+```json
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.only_run_on_ml_node": true,
+    }
+}
+```
+{% include copy-curl.html %}
+
+If you are running a remote inference or local model, you can set `"plugins.ml_commons.only_run_on_ml_node"` to `false` and use data nodes instead.
+
+
+### Setting up connector access control
+
+To enable access control on the connector API, use the following cluster setting:
+
+```json
+PUT /_cluster/settings
+{
+    "persistent": {
+        "plugins.ml_commons.connector_access_control_enabled": true
+    }
+}
+```
+{% include copy-curl.html %}
+
+When enabled, the `backend_roles`, `add_all_backend_roles`, or `access_model` options are required in order to use the connector API. If successful, OpenSearch returns the following response:
+
+```json
+{
+  "acknowledged": true,
+  "persistent": {
+    "plugins": {
+      "ml_commons": {
+        "connector_access_control_enabled": "true"
+      }
+    }
+  },
+  "transient": {}
+}
+```
+
+## Creating a connector
+
+You can build connectors in two ways:
+
+- A **standalone connector**, saved in a connector index, which can be reused and shared with multiple remote models but requires access to both the model and the third-part being accessed by the connector, such as OpenAI.
+
+- An **internal connector**, saved in the model index and which can only be used with one remote model. Unlike a standalone connect, users only need access to the model itself to access the connector, since the connection is made inside the model.
```

Comment:
```suggestion
- An **internal connector**, saved in the model index, can only be used with one remote model. Unlike a standalone connector, users only need access to the model itself to access an internal connector because the connection is established inside the model.
```